### ISSUE TITLE
Flag webhook-timeout provided but not defined for Chart spark-operator-1.1.12

### ISSUE THREAD

**Issue by Taqas** [kubeflow/spark-operator#1392]
When pulling the Helm Chart from https://googlecloudplatform.github.io/spark-on-k8s-operator and installing the latest (=spark-operator-1.1.12) I run into 
`flag provided but not defined: -webhook-timeout`

Most likely because of [this](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/commit/6939cbd887adda19984ed5f75fd49dcbfc7c09b4) commit.

When I install the Chart from [source](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/tree/spark-operator-chart-1.1.12/charts/spark-operator-chart) it works without problems.
```
helm list
NAME      	NAMESPACE     	REVISION	UPDATED                             	STATUS  	CHART                	APP VERSION
my-release	spark-operator	1       	2021-11-15 00:28:23.264778 +0100 CET	deployed	spark-operator-1.1.12	v1beta2-1.2.3-3.1.1
spark     	spark-operator	2       	2021-11-15 00:29:41.35789 +0100 CET 	deployed	spark-operator-1.1.12	v1beta2-1.2.3-3.1.1

k get pods
NAME                                         READY   STATUS             RESTARTS   AGE
my-release-spark-operator-665d77b858-nr9bf   1/1     Running            0          11m
spark-spark-operator-7cccb4f59f-swr22        0/1     CrashLoopBackOff   7          12m
spark-spark-operator-webhook-init-sb4q5      0/1     Completed          0          10m
```

I have updated my repo
```
helm repo add spark-operator https://googlecloudplatform.github.io/spark-on-k8s-operator --force-update
helm repo update
```

Where is the discrepancy coming from?

**Comment by felixng-umh** [comment]
same here

**Comment by j-unior** [comment]
+1

**Comment by pdrastil** [comment]
It's coming from this PR that modified the chart https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1387 and but it seems `gcr.docker.prod.walmart.com/spark-operator/spark-operator:latest` was not published into Docker registry with this feature. Same applies to other image tags.

**Comment by echang0929** [comment]
+1

**Comment by DiogoReisPinto** [comment]
+1 

Any idea when this will get fixed?

**Comment by duyet** [comment]
Currently, I'm working around this by building the spark-operator binary from `master` and adding the current image:
https://github.com/duyet/spark-docker/blob/master/spark-operator/v1beta2-1.2.3-3.1.1-master/Dockerfile

```Dockerfile
ARG SPARK_IMAGE=gcr.io/spark-operator/spark-operator:v1beta2-1.2.3-3.1.1

# Build spark-operator binary from master branch
FROM golang:1.15.2-alpine as builder
WORKDIR /workspace
RUN apk update && apk add git && \
    git clone --depth 1 https://github.com/GoogleCloudPlatform/spark-on-k8s-operator && \
    cd spark-on-k8s-operator && \
    go mod download && \
    CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o /usr/bin/spark-operator main.go

FROM ${SPARK_IMAGE}
USER root
COPY --from=builder /usr/bin/spark-operator /usr/bin/
```

Quickly test this by using my image when installing the operator:

```bash
helm repo add spark-operator https://googlecloudplatform.github.io/spark-on-k8s-operator
helm install <release-name> spark-operator/spark-operator \
  --set image.repository=duyetdev/spark-operator \
  --set image.tag=v1beta2-1.2.3-3.1.1-master
```

**Comment by korjek** [comment]
@liyinan926 could you please help with publishing docker image that contains the most recent changes?

**Comment by autarchprinceps** [comment]
Perhaps you should also set the actual image version in the helm chart, then a change in the helm chart would be synchronous with the chang in the image, you know like any other helm chart does.

**Comment by shivakumargowda12** [comment]
Is this issue fixed in latest release 1.1.15 ? Because, I am still getting same error with master branch also.

**Comment by hobbsh** [comment]
> Is this issue fixed in latest release 1.1.15 ? Because, I am still getting same error with master branch also.

No, it's still broken. And with this version you still need to manually set the image tag as the one set in `appVersion` is not found.

**Comment by apreethi13** [comment]
> a workaround that worked for me and for @tomarad is to specify both image.tag and version in the command line when running helm install `helm install spark-operator spark-operator/spark-operator --namespace data --set sparkJobNamespace=data --set webhook.enable=true --set image.tag=v1beta2-1.2.3-3.1.1 --version=1.1.11`
> 
> the version is the last version before the webhook.timeout flag was added, as seen in [this commit](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/1387/files/43c9a82550d7af1d52e739ac3dfc0f65f47ab4a2)

Bumping down the version from `1.1.13` to `1.1.11` fixed the issue. But would be great if we could have a fix for the latest version.
